geode-core/src/test/java/org/apache/geode/test/dunit/rules/LocatorServerStartupRule.java

```diff
@@ -308,30 +278,68 @@ public MemberVM startServerAsEmbededLocator(int index, Properties properties) th
       serverStarter.before();
       return serverStarter;
     });
-    return setMember(index, new MemberVM(server, serverVM, useTempWorkingDir()));
+
+    MemberVM memberVM = new MemberVM(server, serverVM, useTempWorkingDir());
+    occupiedVMs.put(index, memberVM);
+    return memberVM;
   }
 
   public void stopVM(int index) {
     stopVM(index, true);
   }
 
   public void stopVM(int index, boolean cleanWorkingDir) {
-    MemberVM member = members.get(index);
-    // user has started a server/locator in this VM
-    if (member != null) {
-      member.stopMember(cleanWorkingDir);
-    }
-    // user may have used this VM as a client VM
-    else {
-      getVM(index).invoke(() -> MemberStarterRule.disconnectDSIfAny());
+    VMProvider member = occupiedVMs.get(index);
+
+    if (member == null)
+      return;
+
+    member.stopVM(cleanWorkingDir);
+  }
+
+  public ClientVM startClientVM(int index, String username, String password,
```

Comment:
I don't really like this signature but I guess the only way around would be with a `withServerPort` style of builder. For now it's OK though until we get more usage patterns for client VMs.